sdk/src/types/block/output/foundry.rs

```diff
@@ -510,6 +512,11 @@ impl FoundryOutput {
             Ordering::Greater => {
                 // Melt / Burn
 
+                if capabilities.has_capability(TransactionCapabilityFlag::BurnNativeTokens) {
+                    // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
+                    return Err(StateTransitionError::UnsupportedStateTransition);
+                }
```

Comment:
I'm not fully sure this is sufficient as the match case if for both melt and burn, needs to think about it a bit more 🤔

Reply:
Yeah I don't think this is correct, doing a quick test